lib/rules/widget-not-inline-matches.js

```diff
@@ -0,0 +1,30 @@
+import { getRole, getRoleType } from '../commons/aria';
+import { isFocusable, isInTextBlock } from '../commons/dom';
+import svgNamespaceMatches from './svg-namespace-matches';
+
+export default function widgetNotInline(node, vNode) {
+  return matchesFns.every(fn => fn(node, vNode));
+}
+
+const matchesFns = [
+  (node, vNode) => isWidgetType(vNode),
+  (node, vNode) => isNotAreaElement(vNode),
+  (node, vNode) => !svgNamespaceMatches(node, vNode),
+  (node, vNode) => isFocusable(vNode),
+  node => !isInTextBlock(node, { noLengthCompare: true })
+];
+
+function isWidgetType(vNode) {
+  const role = getRole(vNode);
+  if ('option' === role) {
+    return false;
+  } else if (['combobox', 'listbox'].includes(role)) {
```

Comment:
For consideration; I could potentially change the role type of these three so that everything we want is a widget, and everything we don't isn't. I didn't explore what the wider implications of that could be.